### ISSUE TITLE
Can't get the node socket server to work: Unexpected response code: 502 

### ISSUE THREAD

**Issue by emielmolenaar** [REBELinBLUE/deployer#299]
Before submitting your issue, please make sure that you've checked all of the checkboxes below.

- [x] You're running the [latest release](https://github.com/REBElinBLUE/deployer/releases/latest) version of Deployer.
- [x] Ensure that you're running at least PHP 5.5.9, you can check this by running `php -v`
- [x] You've ran `composer install --no-dev` from the root of your installation.
- [x] You've ran `npm install --production` from the root of your installation.
- [x] You've ran `rm -rf bootstrap/cache/*` from the root of your installation.
- [x] You have restarted the queue listener and node socket server.

To help us better understand your issue, please answer the following.

### Expected behaviour

Status reporting on running deployments.

### Actual behaviour

Server error
A connection could not be established to the socket at https://deploy.freelytravel.com. This is required for reporting the status on running deployments. Please reload, if the issue continues please contact the system administrator.

I have configured everything as documented; here's my nginx vhost config:

```
upstream deployer {
        server 127.0.0.1:1337;
    }

server {
    listen 443 ssl http2;
    # * snip *
    # large_client_header_buffers 8 32k;
    ssl_certificate /path/to/fullchain.pem;
    ssl_certificate_key /path/to/privkey.pem;
    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
    ssl_ciphers 'EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH';
    ssl_prefer_server_ciphers on;
    ssl_session_cache shared:SSL:10m;
    ssl_dhparam /etc/ssl/certs/dhparam.pem;
    ssl_stapling on;
    ssl_stapling_verify on;
    resolver 8.8.8.8 8.8.4.4 valid=300s;
    resolver_timeout 5s;
    add_header Strict-Transport-Security "max-age=63072000; includeSubdomains; ";
    add_header X-Frame-Options SAMEORIGIN;
    add_header X-Content-Type-Options nosniff;
    add_header X-XSS-Protection "1; mode=block";

    location / {
     try_files $uri $uri/ /index.php$is_args$args;
    }
    location ~ \.php$ {
      fastcgi_split_path_info ^(.+\.php)(/.+)$;
      fastcgi_pass unix:/var/run/php7.1-fpm.sock;
      fastcgi_index index.php;
      fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
      include fastcgi_params;
    }
    location /socket.io {
      # proxy_buffers 8 32k;
      # proxy_buffer_size 64k;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection "upgrade";
      proxy_http_version 1.1;
      proxy_set_header X-Real-IP $remote_addr;
      proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
      proxy_set_header X-Forwarded-Proto $scheme;
      proxy_set_header Host $host;
      proxy_pass http://deployer;
    }
}
```
I've set up supervisor to run the node process. This works as intended, I can see the process running on the server. I am able to start `node socket.js` in a terminal and it seems to be running.

Relevant .env info:

```
SOCKET_URL=https://nameofmyserver.com
SOCKET_PORT=1337
SOCKET_SSL_KEY_FILE=/path/to/privkey.pem
SOCKET_SSL_CERT_FILE=/path/to/cert.pem
SOCKET_SSL_CA_FILE=/path/to/chain.pem
SOCKET_SSL_KEY_PASSPHRASE=
```

As you can see I am running deployer over HTTPS, we are using Let's Encrypt certificates. This is what nginx says:

`*203 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: 84.86.122.7, server: nameofmyserver.com, request: "GET /socket.io/?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE0ODQwNjcyMTQsImp0aSI6Ik0zTmhVVE5tZEhRMVlqaGtkRGRUUjNGbmNERlVNMlpLV1VWMmRIUk9UMlk9IiwiaXNzIjoiaHR0cHM6XC9cL2RlcGxveS5mcmVlbHl0cmF2ZWwuY29tIiwibmJmIjoxNDg0MDY3MjE0LCJleHAiOjE0ODQwNzgwMTQsImRhdGEiOnsidXNlcklkIjoxfSwic3ViIjoxfQ.FCQ-_rqz68g337mGppZ2Y2Q5dgY3KoYQ5AMFu_kcxQE&EIO=3&transport=websocket HTTP/1.1", upstream: "http://127.0.0.1:1337/socket.io/?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE0ODQwNjcyMTQsImp0aSI6Ik0zTmhVVE5tZEhRMVlqaGtkRGRUUjNGbmNERlVNMlpLV1VWMmRIUk9UMlk9IiwiaXNzIjoiaHR0cHM6XC9cL2RlcGxveS5mcmVlbHl0cmF2ZWwuY29tIiwibmJmIjoxNDg0MDY3MjE0LCJleHAiOjE0ODQwNzgwMTQsImRhdGEiOnsidXNlcklkIjoxfSwic3ViIjoxfQ.FCQ-_rqz68g337mGppZ2Y2Q5dgY3KoYQ5AMFu_kcxQE&EIO=3&transport=websocket"`

What can I do to debug this a bit more? Clearly some communication between nginx and node isn't working out, but I am unable to locate the problem so far. Any help would be appreciated. Thanks!

### Environment info
Operating System: Debian 8.6 x64
PHP Version: 7.1
Node Version: 6.9.4
Database System: mariadb
Database Version: 10.1

**Comment by REBELinBLUE** [comment]
Could you set `APP_DEBUG` to `true` in the `.env` file and then run the node server manually? Then log in and see if there is the node server outputs anything to the terminal.

If you use chrome, open the developer console and look at the network tab, then the "ws" tab then see if there are any attempts to connect (you will need the developer console open before loading the page)

**Comment by emielmolenaar** [comment]
I have set everything to debug. No console output unfortunately. 

So, I just removed the whole HTTPS part in socket.js and fired everything up. Everything was working now. The HTTPS stuff is causing trouble clearly. 

First thing that popped into my mind, was ofcourse the mismatching domain names between the upstream and de SSL certificate (https://127.0.0.1 vs https://nameofmyserver.com) ... After setting up the upstream to https://nameofmyserver.com, the problem was solved.

So, quite obviously, when using HTTPS the domain name of the upstream should matcht the certificate's one. A small hint on this in the docs might help other people (who have been staring at their screen for too long too) in the future?

Thanks!
